hydra/_internal/core_plugins/importlib_resources_config_source.py

```diff
@@ -1,13 +1,28 @@
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 import os
+import sys
 from typing import List, Optional
 
-import importlib_resources
 from omegaconf import OmegaConf
 
 from hydra.core.object_type import ObjectType
 from hydra.plugins.config_source import ConfigLoadError, ConfigResult, ConfigSource
 
+if sys.version_info[0] == 3 and sys.version_info[1] < 9:
+    import importlib_resources
+
+    # `importlib_resources` is avilable only till Python 3.8. Beyond this,
+    # one should use `importlib.resources`.
+    # Note that `importlib.resources` is available on Python 3.8 but its
+    # API does not match the API of `importlib_resources`. However, in
+    # Python 3.9, the API of `importlib.resources` matches the API of
+    # `importlib_resources`.
+    # So, by default, we try to use `importlib_resources` (supported till 3.8)
+    # and if that is not found, we use `importlib.resources`.
+    # Switching the order of import statements breaks the code for Python 3.8.
```

Comment:
This comment with both very verbose and inaccurate.
The logic one is correct and does not need justification:
importlib_resources was always meant to be used until Python 3.9.
read about it [here](https://importlib-resources.readthedocs.io/en/latest/).

Hydra is not exposed to importlib directly other than through this config source.

1. swap the order to the order I suggested:
```
if python >= 3.9:
  use built in
else:
  use backport
```

2. change the comment:
```suggestion
    # Use importlib backport for Python older than 3.9
```